src/chocolatey/infrastructure.app/commands/ChocolateyApiKeyCommand.cs

```diff
@@ -157,15 +181,16 @@ public virtual void noop(ChocolateyConfiguration configuration)
 
         public virtual void run(ChocolateyConfiguration configuration)
         {
-            if (configuration.ApiKeyCommand.Remove)
+            switch (configuration.ApiKeyCommand.Command)
             {
-                this.Log().Warn(ChocolateyLoggers.Important, "DEPRECATION NOTICE");
-                this.Log().Warn(RemoveOptionDeprecationMessage);
-                _configSettingsService.remove_api_key(configuration);
-            }
-            else if (string.IsNullOrWhiteSpace(configuration.ApiKeyCommand.Key))
-            {
-                _configSettingsService.get_api_key(configuration, (key) =>
+                case ApiKeyCommandType.Remove:
+                    _configSettingsService.remove_api_key(configuration);
+                    break;
+                case ApiKeyCommandType.Add:

```

Comment:
Does this break anyone doing `choco apikey -s kjdf -k sdlkfj`.  We haven't signalled anywhere in deprecation notices that this would be going away.

Reply:
Just reading the description of the PR, and sounds like it does! 👍🏻 I will take a closer look and take this for a spin tomorrow, as I have only had a quick review of the PR just now.

Reply:
Yeah I have it default to `List` mode when it does argument validation / extra argument processing like `choco source` does, and if they provide at least `-k` / `--api-key` then I assume they're trying to `Add` instead and go with that assumption.